portal-api/src/scheduler/jobs/helpers/utilisation-report-helpers.js

```diff
@@ -0,0 +1,47 @@
+const { startOfMonth, format, subMonths } = require('date-fns');
+const externalApi = require('../../../external-api/api');
+const { addBusinessDaysWithHolidays } = require('../../../utils/date');
+
+/**
+ * Returns the utilisation report due date for the current month
+ * @returns {Promise<Date>}
+ */
+const getReportDueDate = async () => {
+  const bankHolidays = await externalApi.bankHolidays.getBankHolidayDatesForRegion('england-and-wales');
+  return addBusinessDaysWithHolidays(startOfMonth(new Date()), process.env.PORTAL_REPORTING_DUE_DATE_BUSINESS_DAYS_FROM_START_OF_MONTH, bankHolidays);
+};
+
+/**
+ * Returns the utilisation report due date for the current month in 'do MMMM yyyy' format
+ * @returns {Promise<string>}
+ */
+const getFormattedReportDueDate = async () => {
+  const reportDueDate = await getReportDueDate();
+  return format(reportDueDate, 'do MMMM yyyy');
+};
+
+/**
+ * Returns the month (1-indexed) and year of the current report period (i.e. the previous month)
+ * @returns {{ month: number, year: number }}
+ */
+const getReportPeriodMonthAndYear = () => {
+  const lastMonthDate = subMonths(startOfMonth(new Date()), 1);
```

Comment:
I don't dislike the use of `date-fns`, but I'm curious as to why this is better than something like:
```
const lastMonth = new Date().getMonth() - 1;
const oneIndexedMonth = lastMonth + 1;
```

Reply:
I can update this to remove `startOfMonth` though as that's not needed. This was used to handle an edge case where the day of the current month were more than the new previous month. `subMonths` is clever enough to handle this case though. e.g. if I do
```
const lastMonthDate = subMonths(new Date('2023-03-31'), 1);
```
it will return a `Date('2023-02-28')` and not fail, or instead do anything weird like return `Date('2023-03-03')`.